provider/resources.go

```diff
@@ -626,6 +615,29 @@ func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) erro
 	return nil
 }
 
+// We should only run the validation once to avoid duplicating the reported errors.
+var credentialsValidationOnce sync.Once
+
+// preConfigureCallback validates that AWS credentials can be successfully discovered. This emulates the credentials
+// configuration subset of `github.com/terraform-providers/terraform-provider-aws/aws.providerConfigure`.  We do this
+// before passing control to the TF provider to ensure we can report actionable errors.
+func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) error {
+	skipCredentialsValidation := boolValue(vars, "skipCredentialsValidation",
+		[]string{"AWS_SKIP_CREDENTIALS_VALIDATION"})
+
+	// if we skipCredentialsValidation then we don't need to do anything in
+	// preConfigureCallback as this is an explicit operation
+	if skipCredentialsValidation {
+		return nil
+	}
+
+	var err error = nil
```

Comment:
The zero value is implicit.
```suggestion
	var err error
```